target/linux/kirkwood/files-5.4/arch/arm/boot/dts/kirkwood-nsa310s.dts

```diff
@@ -0,0 +1,231 @@
+/dts-v1/;
+
+#include "kirkwood-nsa3x0-common.dtsi"
+
+/ {
+	model = "ZyXEL NSA310S";
+	compatible = "zyxel,nsa310s", "marvell,kirkwood-88f6702", "marvell,kirkwood";
+
+	aliases {
+		led-boot = &led_green_sys;
+		led-failsafe = &led_orange_sys;
+		led-running = &led_green_sys;
+		led-upgrade = &led_orange_sys;
+	};
+
+	memory {
+		device_type = "memory";
+		reg = <0x00000000 0x10000000>;
+	};
+
+	chosen {
+		bootargs = "console=ttyS0,115200n8 earlyprintk";
+		stdout-path = &uart0;
+	};
+
+	ocp@f1000000 {
+		pinctrl: pin-controller@10000 {
```

Comment:
This whole block is already in kirkwood-nsa3x0-common.dtsi, but with different pins. 

I wonder whether it would actually be easier to not derive from it, but to add the relevant definitions here (instead of overwriting most of them partially).

Reply:
I will consider drop using dtsi. Dts introduces many changes.